packages/template-retail-react-app/app/components/store-locator-modal/index.jsx

```diff
@@ -7,60 +7,41 @@
 
 import React, {useState} from 'react'
 import PropTypes from 'prop-types'
-import {useIntl} from 'react-intl'
 import {
     Modal,
     ModalBody,
     ModalCloseButton,
     ModalContent,
     useBreakpointValue
 } from '@salesforce/retail-react-app/app/components/shared/ui'
-import StoreLocatorContent from '@salesforce/retail-react-app/app/components/store-locator-modal/store-locator-content'
-import {useSearchStores} from '@salesforce/commerce-sdk-react'
 import {
-    DEFAULT_STORE_LOCATOR_COUNTRY_CODE,
+    DEFAULT_STORE_LOCATOR_COUNTRY,
     DEFAULT_STORE_LOCATOR_POSTAL_CODE,
-    STORE_LOCATOR_DISTANCE
+    STORE_LOCATOR_NUM_STORES_PER_LOAD
 } from '@salesforce/retail-react-app/app/constants'
-import {noop} from '@salesforce/retail-react-app/app/utils/utils'
-import {useForm} from 'react-hook-form'
+import StoreLocatorContent from '@salesforce/retail-react-app/app/components/store-locator-modal/store-locator-content'
 
-const StoreLocatorModal = ({isOpen, onClose = noop}) => {
-    const intl = useIntl()
+const StoreLocatorModal = ({isOpen, onClose}) => {
+    const [userHasSetManualGeolocation, setUserHasSetManualGeolocation] = useState(false)
 
     const [searchStoresParams, setSearchStoresParams] = useState({
```

Comment:
What was the reason for this state to be defined in this component? `StoreLocatorModal` does not use it for its rendering. It's simply passing the state to another component.

How about defining the state directly in `StoreLocatorContent` component instead?

Reply:
The `userHasSetManualGeolocation` and `searchStoresParams` are defined in this component because if it is inside `StoreLocatorModal` which gets conditionally loaded on and off, the states will be reset when the store locator is closed, meaning the user input of the store locator will not be saved.